tests/e2e/provisioning/pkg/client/broker/broker_client.go

```diff
@@ -0,0 +1,229 @@
+package broker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/thanhpk/randstr"
+
+	"github.com/google/uuid"
+
+	"github.com/pivotal-cf/brokerapi/v7/domain"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type Config struct {
+	URL  string
+	Auth struct {
+		Username string
+		Password string
+	}
+	ProvisionGCP     bool
+	ProvisionTimeout time.Duration
+}
+
+type Client struct {
+	brokerConfig    Config
+	clusterName     string
+	runtimeID       string
+	globalAccountID string
+
+	client *http.Client
+	log    logrus.FieldLogger
+}
+
+func NewClient(config Config, globalAccountID, runtimeID string, clientHttp http.Client, log logrus.FieldLogger) *Client {
+	return &Client{
+		brokerConfig:    config,
+		runtimeID:       runtimeID,
+		clusterName:     fmt.Sprintf("%s-%s", "e2e-provisioning", strings.ToLower(randstr.String(10))),
+		globalAccountID: globalAccountID,
+		client:          &clientHttp,
+		log:             log,
+	}
+}
+
+const (
+	kymaClassID = "47c9dcbf-ff30-448e-ab36-d3bad66ba281"
+	gcpPlanID   = "ca6e5357-707f-4565-bbbd-b3ab732597c6"
+	azurePlanID = "4deee563-e5ec-4731-b9b1-53b42d855f0c"
+
+	instancesURL = "/v2/service_instances"
+)
+
+func (c *Client) ProvisionRuntime() (string, error) {
+	c.log.Infof("Provisioning Runtime [ID: %s, NAME: %s]", c.runtimeID, c.clusterName)
+	requestByte, err := c.prepareProvisionDetails()
+	if err != nil {
+		return "", errors.Wrap(err, "while marshalling request body")
```

Comment:
```suggestion
		return "", errors.Wrap(err, "while preparing provision details")
```

you do not know why this operation failed ;) what's more in current impl you will end up with log:
```
"while marshalling request body: while marshalling request body:  <error_msg>"
```